zerver/lib/email_notifications.py

```diff
@@ -117,6 +118,54 @@ def make_emoji_img_elem(emoji_span_elem: CSSSelector) -> Dict[str, Any]:
     content = lxml.html.tostring(fragment).decode('utf-8')
     return content
 
+def fix_spoilers_in_html(content: str, language: str) -> str:
+    with override_language(language):
+        spoiler_title: str = _("Open Zulip to see the spoiler content")
+    fragment = lxml.html.fromstring(content)
+    spoilers = fragment.find_class("spoiler-block")
+    for spoiler in spoilers:
+        header = spoiler.find_class("spoiler-header")[0]
+        spoiler_content = spoiler.find_class("spoiler-content")[0]
+        header_content = header.find("p")
+        if header_content is None:
+            # Create a new element to append the spoiler to)
+            header_content = lxml.html.fromstring("<p></p>")
+            header.append(header_content)
+        else:
+            # Add a space. Its simpler to append a new span element than
+            # inserting text after the last node ends since neither .text
+            # and .tail do the right thing for us.
+            header_content.append(lxml.html.fromstring("<span> </span>"))
+        span_elem = lxml.html.fromstring(
+            f'<span class="spoiler-title" title="{spoiler_title}">({spoiler_title})</span')
+        header_content.append(span_elem)
+        header.drop_tag()
```

Comment:
Maybe add some comments explaining what this function does. Its name suggests that it removes the whole content of the tag while it just adds the content to parent in-place.